ts/session/types/OpenGroup.ts

```diff
@@ -55,17 +77,130 @@ export class OpenGroup {
       conversationId,
     } as OpenGroupParams;
 
-    if (this.serverRegex.test(server)) {
+    const isValid = OpenGroup.serverRegex.test(server);
+    if (!isValid) {
       return;
     }
 
     return new OpenGroup(openGroupParams);
   }
 
-  private static getServer(groupId: string): string | undefined {
+  /**
+   * Join an open group
+   *
+   * @param server The server URL
+   * @param onLoading Callback function to be called once server begins connecting
+   * @returns `OpenGroup` if connection success or if already connected
+   */
+  public static async join(
+    server: string,
+    onLoading?: any
+  ): Promise<OpenGroup | undefined> {
+    const prefixedServer = OpenGroup.prefixify(server);
+    if (!OpenGroup.validate(server)) {
+      return;
+    }
+
+    // Make this not hard coded
+    const channel = 1;
+    let conversation;
+    let conversationId;
+
+    // Return OpenGroup if we're already connected
+    conversation = await OpenGroup.getConversation(prefixedServer);
+    if (conversation) {
+      conversationId = conversation?.cid;
+      if (conversationId) {
+        return new OpenGroup({
+          server: prefixedServer,
+          channel: 1,
+          conversationId,
+        });
+      }
+    }
+
+    // Try to connect to server
+    try {
+      if (onLoading) {
+        onLoading();
+      }
+
+      conversation = await window.attemptConnection(prefixedServer, channel);
+      conversationId = conversation?.cid;
+    } catch (e) {
+      throw new Error(e);
+    }
+
+    // Do we want to add conversation as a property of OpenGroup?
+    return new OpenGroup({
+      server,
+      channel,
+      conversationId,
+    });
+  }
+
+  /**
+   * Get the conversation model of a server from its URL
+   *
+   * @param server The server URL
+   * @returns BackBone conversation model corresponding to the server if it exists, otherwise `undefined`
+   */
+  public static async getConversation(server: string): Promise<any> {
+    if (!OpenGroup.validate(server)) {
+      return;
+    }
+
+    const prefixedServer = this.prefixify(server);
+    const serverInfo = (await window.lokiPublicChatAPI.findOrCreateServer(
+      prefixedServer
+    )) as any;
+
+    if (!serverInfo?.channels?.length) {
+      return;
+    }
+
+    return serverInfo.channels[0].conversation;
+  }
+
+  /**
+   * Get the conversation model of a server from conversation ID
+   *
+   * @param conversationId The server's conversation ID
+   * @returns BackBone conversation model corresponding to the server if it exists, otherwise `undefined`
+   */
+  public static getConversationByCID(conversationId: string): any {
```

Comment:
```suggestion
  public static getConversationByID(conversationId: string): any {
```
there is a cid field on the conversation, it's not the want to use right?